### ISSUE TITLE
Maximum message payload size

### ISSUE THREAD

**Issue by kevinmehall** [eProsima/Fast-RTPS#20]
I tested sending large messages and found the behavior inconsistent by message size and platform. It seems there is a known limitation at approximately 64k bytes, and potentially some other issues below that. What would need to be changed to be able to send 300KB or larger messages?

Here's what I saw in my tests on OSX and Linux:

Larger than ~8650 bytes on OSX: messages appear to send successfully in Release builds, but are not received. Warnings are printed Debug builds:
```
[Info][SendResource::sendSync]                   UDPv4: 8716 bytes TO endpoint: 10.37.129.2:27411 FROM 10.37.129.2:10041
[Warning][SendResource::sendSync]                   Error: send_to: Message too long
[Info][SendResource::sendSync]                   SENT 0
```

Larger than ~16000 bytes on OSX and Linux: sender error
```
[Error][RTPSMessageGroup::prepareDataSubM]        Problem adding DATA submsg to the CDRMessage, buffer too small
```

Larger than 64000 bytes: sender error on initialization
```
[Error][ParticipantImpl::registerType]            Current version only supports types of sizes < 64000
```
Larger than 65536 bytes: compiler warning on uint16_t overflow.

**Comment by liamstask** [comment]
perhaps relatedly, when I run the `LatencyTest` example (in `examples/C++/LatencyTest`) on my OS X machine, it stalls after the `4096` round and does not print any output for the `8192` round.

**Comment by richiprosima** [comment]
I've updated branch [hotfix/18](https://github.com/eProsima/Fast-RTPS/tree/hotfix/18). It contains the code we are testing to be release 1.0.6. I've added a test to check your problem. This test sends a sequence of 63996 bytes without any problem on my Linux and Windows. Also in a OSX virtualmachine.

If you provide us with your code we will check it. Are you using the
participant qos ```PParam.rtps.sendSocketBufferSize```?

Data larger than 64kb is not supported right now. We are working to support
this feature. Meanwhile we provide you an example to send this larger data,
fragmenting it. This example will be in github with release 1.0.6 but we've
uploaded it
[here](https://gist.github.com/RafaelLaraAlvarez/eba95cc4eceede79f6b3).
